tests/console/validate_raid.pm

```diff
@@ -0,0 +1,146 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Simple RAID partitioning layout validation
+# Maintainer: Joaquín Rivera <jeriveramoya@suse.com>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use version_utils 'is_sle';
+use Test::Assert ':all';
+
+#
+# Define blocks of expected data for partitioning layout in different products and architectures
+#
+# 4 btrfs partitions mounted on / in raid partition
+my $btrfs = qr/(md0(p1)?.+btrfs.+\/.*){4}/s;
+# 4 swap partitions in raid partition
+my $swap = qr/(md((1|2)p1)?.+swap.+\[SWAP\].*){4}/s;
+# 8 raid partitions
+my $raid_partitions_2_arrays = qr/(md(0|1).*){8}/s;
+# 12 raid partitions
+my $raid_partitions_3_arrays = qr/(md(0|1|2).*){12}/s;
+# 8 linux raid members
+my $linux_raid_member_2_arrays = qr/(vd(a|b|c|d)(2|3).+linux_raid_member.*){8}/s;
+# 12 linux raid members
+my $linux_raid_member_3_arrays = qr/(vd(a|b|c|d)(1|2|3|4).+linux_raid_member.*){12}/s;
+# 4 hard disks
+my $hard_disks = qr/(vd(a|b|c|d)\D+.*){4}/s;
+# 4 ext4 partitions mounted on /boot
+my $ext4_boot = qr/(md1(p1)?.+ext4.+\/boot.*){4}/s;
+# Unique vfat partition in first disk (mounted on /boot)
+my $vfat_efi = qr/vda1.+vfat.\s+\/boot\/efi.*(vd(b|c|d)1(?!.*vfat).*){3}/s;
+#
+# Define blocks of expected data for raid configuration in different products and architectures
+#
+# RAID arrays
+my @raid_arrays = qw(/dev/md0 /dev/md1);
+# Number of array devices
+my $num_raid_arrays = @raid_arrays;
+# RAID level from settings
+my $level = get_required_var('RAIDLEVEL');
+# RAID array with corresponding RAID level
+my $raid_level = qr/\/dev\/md0:.*?Raid Level : raid$level/s;
```

Comment:
Ok, our idea with named regexp will require too many changes to the PR. But for your info, here is how it could look like.
```perl
my $raid_level = qr/\/dev\/md0:.*Raid Level : raid(?<raid_level>\d+)/s;
 assert_matches($level, $+{raid_level}, "Raid level check for md0 failed");
```

Reply:
Thanks for those cases it would work but most of the regex are taking into account repetition. I didn't find in perl the way to capture in cases like this:
`(md(?<number>\d).*){8}` Basically .* matches until the end of the full output, the break line is treated as another character with /s modifier and not sure how to call it later in case it would be possible unless hash $+{number} could contain an array, but this is also bad example as "md" would remain out of the match, but just to explain what I found tricky. We'll figure out all together, thanks! :)

Reply:
Right, we don't have answer yet, but your PR shows us that there is a question we need to answer =)